### PR TITLE
Created the product model and added it as a dbset to the congo context.

### PR DESCRIPTION
**Pull request by IliaIliev94**
Closes Issue #15.

### CODE REVIEW

**Comment by ThorstenSauter** [comment]
Should we generate the value of the `Id` property in C# or have the database provide it automatically, like it already would by default (according to the [docs](https://docs.microsoft.com/en-us/ef/core/modeling/generated-properties?tabs=data-annotations#primary-keys))?

This would result in the following change:
```diff
[Key]
- public string Id { get; init; } = Guid.NewGuid().ToString();
+ public string Id { get; set; }
```

**Comment by IliaIliev94** [comment]
> Should we generate the value of the `Id` property in C# or have the database provide it automatically, like it already would by default (according to the [docs](https://docs.microsoft.com/en-us/ef/core/modeling/generated-properties?tabs=data-annotations#primary-keys))?
> 
> This would result in the following change:
> ```diff
> [Key]
> - public string Id { get; init; } = Guid.NewGuid().ToString();
> + public string Id { get; set; }
> ```

I made it using a Guid, because if the Id is randomly generated every time it would be harder for users to get to a product page by just changing the Id parameter in the url (like changing /products/1 to products/2 in the url), which is sometimes an unwanted side effect. If the other members decide that we should use an Id automatically provided by the database, then I can change it.

**Comment by ThorstenSauter** [comment]
Ah, that's my bad for not paying attention to the actual type of `Id`. I just assumed it was `Guid` from the initial value. I support using a `Guid` for the `Id`, but if we actually use `Guid` as the type, the database can generate it, as I alluded to earlier.
This time hopefully correct:
```diff
- public string Id { get; init; } = Guid.NewGuid().ToString();
+ public Guid Id { get; set; }
```

File: src/Congo.WebApi/Data/CongoContext.cs [hunk 1]
```diff
@@ -1,9 +1,11 @@
-﻿using Microsoft.EntityFrameworkCore;
+﻿using Congo.WebApi.Data.Models;
+using Microsoft.EntityFrameworkCore;
 
 namespace Congo.WebApi.Data
 {
     public class CongoContext : DbContext
     {
+        public DbSet<Product> Products { get; set; }
```
Comment (pobiega) [inline src/Congo.WebApi/Data/CongoContext.cs]: I think we should follow standard C# structure here, which would be to put properties below the constructor.

File: src/Congo.WebApi/Data/Models/Product.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Congo.WebApi.Data.Models
+{
+    public class Product
+    {
+        [Key]
+        public string Id { get; init; } = Guid.NewGuid().ToString();
```
Comment (pobiega) [inline src/Congo.WebApi/Data/Models/Product.cs]: Agree with Thorsten, if we want Guid style IDs, this should be `Guid` prop, not a string. My only gripe with them is that they are very long and take up a lot of URL space, so an alternative worth considering is the "YouTube" style URL-safe base64 IDs. Doable, but perhaps overkill.

I will not die on this hill, so unless you feel very strongly about making a YouTube style identifier, Guid is fine.

File: src/Congo.WebApi/Data/Models/Product.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+        [Required]
```
Comment (pobiega) [inline src/Congo.WebApi/Data/Models/Product.cs]: I'm not a big fan of `DataAnnotations` to be honest. If the desired result is to make this not nullable, I suggest we use `IEntityTypeConfiguration<T>` instead. Here is the doc page: https://docs.microsoft.com/en-us/ef/core/modeling/

It keeps the domain object pure and more POCO, and keeps the database-specific stuff away from it.

This goes for `[Key]` as well, of course.

**Review by pobiega — CHANGES_REQUESTED** [review]

File: src/Congo.WebApi/Data/CongoContext.cs [hunk 4]
```diff
@@ -1,13 +1,22 @@
...
-        public DbSet<Product> Products { get; set; }
         public CongoContext(DbContextOptions<CongoContext> options) : base(options)
         {
         }
+
+        public DbSet<Product> Products { get; set; }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            new ProductEntityTypeConfiguration().Configure(modelBuilder.Entity<Product>());
```
Comment (pobiega) [inline src/Congo.WebApi/Data/CongoContext.cs]: Can we use the assembly scanning version instead here? That way we don't need to explicitly add another entry for each configuration we set up.

```csharp
modelBuilder.ApplyConfigurationsFromAssembly(typeof(CongoContext).Assembly);
```

**Review by pobiega — CHANGES_REQUESTED** [review]

**Review by pobiega — APPROVED** [review]

**Pull request merged**